WcaOnRails/app/models/competition_payment_integration.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+# TODO: Add validation to ensure that multiple accounts of the same type don't get added to CompetitionPaymentIntegration
+class CompetitionPaymentIntegration < ApplicationRecord
+  belongs_to :connected_account, polymorphic: true
+
+  belongs_to :competition
+
+  # enum available_integrations: {
```

Comment:
Was never able to get the enum working, and wasn't able to find any documentation/code snippets demonstrating your usage - so just went with a hash and scopes, but happy to refactor it!

Reply:
Seems like using an enum breaks polymorphic class loading. Will file an issue against Rails for this

Reply:
Rails is of the opinion that this "shouldn't be a supported use-case for enums". And when Rails has an opinion, you don't argue. https://github.com/rails/rails/issues/17844